shared/api/auth.go

```diff
@@ -105,6 +105,41 @@ type IdentityPut struct {
 	Groups []string `json:"groups" yaml:"groups"`
 }
 
+// IdentitiesPostTLS contains required information for the creation of a TLS identity.
+//
+// swagger:model
+//
+// API extension: access_management_tls.
+type IdentitiesPostTLS struct {
+	// Name associated with the identity
+	// Example: foo
+	Name string `json:"name" yaml:"name"`
+
+	// Trust token (used to add an untrusted client)
+	// Example: blah
+	TrustToken string `json:"trust_token" yaml:"trust_token"`
+
+	// Whether to create a certificate add token
+	// Example: true
+	Token bool `json:"token" yaml:"token"`
+
+	// The PEM encoded and base64 encoded public certificate of the identity
+	Certificate string `json:"certificate" yaml:"certificate"`
+
+	// Groups is the list of groups for which the identity is a member.
+	// Example: ["foo", "bar"]
+	Groups []string `json:"groups" yaml:"groups"`
+}
+
+// IdentityTokenTLS contains a token that can be used by an untrusted client to gain trust with a LXD system.
+//
+// swagger:model
+//
+// API extension: access_management_tls.
+type IdentityTokenTLS struct {
```

Comment:
whats this for?

Reply:
This is what is returned by `POST /1.0/auth/identities/tls` when `{"token": true}` is received by a sufficiently privileged identity.

Now that I think about it, we can just return the `CertificateAddToken` as `POST /1.0/certificates` does. The client can base64 encoded the token before printing it.

Reply:
yeah thats better because we leave the token presentation (base64 encoding it) to the CLI

Reply:
Done.